datafusion/src/logical_plan/builder.rs

```diff
@@ -162,14 +162,15 @@ impl LogicalPlanBuilder {
     ) -> Result<Self> {
         let table_schema = Arc::new(table_schema.clone());
         let provider = Arc::new(EmptyTable::new(table_schema));
-        Self::scan(name.unwrap_or(UNNAMED_TABLE), provider, projection)
+        Self::scan(name.unwrap_or(UNNAMED_TABLE), provider, projection, None)
     }
 
     /// Convert a table provider into a builder with a TableScan
     pub fn scan(
         table_name: impl Into<String>,
         provider: Arc<dyn TableProvider>,
         projection: Option<Vec<usize>>,
+        filters: Option<Vec<Expr>>,
```

Comment:
I think this argument is likely going to  be confusing to users and it should be removed. 

For example as a user of `LogicalPlanBuilder` I would probably assume that the following plan would return only rows where with `a<5` 

```rust
  // Build a plan that looks like it would filter out all rows with `a < 5`
  let plan = builder.scan("table", provider, None, vec![col("a").lt(lit(5)));
```

However, I am pretty sure it could (and often would) return rows with a >= 5). This is because `filters`  added to a `TableScan` node are optional (in the sense that the provider might not filter rows that do not pass the predicate, but is not required to). Indeed, even for the parquet provider, the filters are only used for row group pruning which may or may not be able to filter rows.

I think we could solve this with:
1. Leave `scan` signature alone and rely on the predicate pushdown optimization to push filters appropriately down to the scan (my preference as it is simpler for the users)
2. Rename this argument to something like 'optional_filters_for_performance'  and document what it does more carefully. I think it would be challenging to explain as it might/might not do anything depending on how the data was laid out.

Reply:
Removed, and keep the filters not deserialized for ballista as before.